### PR TITLE
Add GitHub Copilot hook support (CLI + cloud agent)

### PR DESCRIPTION
**Pull request by pbakaus**
## What

Wires the Impeccable design detector into **GitHub Copilot's hook system**, so direct file edits in Copilot get the same post-edit design feedback that Claude Code, Codex, and Cursor already provide. Covers both the **Copilot CLI** and the **cloud/app agent** via the repo-level manifest at `.github/hooks/impeccable.json`.

**Live-verified** against Copilot CLI 1.0.63: a trusted interactive edit fires the hook and returns the expected design reminder (see "Verification" below).

## How it works (captured from the real CLI, not just docs)

Copilot's hook contract differs from the existing harnesses, so I captured real events from the installed CLI:

| Aspect | Copilot | Claude/Codex |
|---|---|---|
| Manifest | `.github/hooks/impeccable.json` (committed; CLI honors it once trusted + on the default branch, cloud agent reads the repo) | `.claude/settings.json`, `.codex/hooks.json` |
| Entry shape | flat `postToolUse` with `type`/`matcher`/`bash`/`timeoutSec` | nested `{matcher, hooks:[{command, timeout}]}` |
| Matcher | full-match regex `edit|create|apply_patch` | `Edit\|Write\|MultiEdit` |
| Stdin event | camelCase `{sessionId, cwd, toolName, toolArgs, toolResult}` | snake_case `{tool_name, tool_input, …}` |
| `toolArgs` | JSON **string** for `edit`/`create`/`view`; raw **patch string** for `apply_patch` | object |
| Output | top-level `{ "additionalContext": "…" }` | `hookSpecificOutput.additionalContext` |

**Tool-name gotcha (caught by the live test):** `copilot -p` edits via `edit`/`create` (JSON `toolArgs` with the path under `path`), but **interactive sessions and the cloud agent edit via `apply_patch`**, whose `toolArgs` is a raw OpenAI-format patch string (`*** Begin Patch` / `*** Add File:`). The matcher and runtime cover all three; `apply_patch` reuses the existing `parseApplyPatchPaths`/`resolveTargetFiles` plumbing (same format Codex uses).

## Changes

- **`scripts/lib/transformers/hooks.js`** — `buildGitHubHooksManifest()` + `github` route in `hooksJsonFor()`.
- **`scripts/lib/transformers/providers.js`** — `emitHooks` / `hooksManifestRel` for the github provider.
- **`skill/scripts/hook-lib.mjs`** — detect the github harness; normalize the camelCase event, routing `apply_patch`'s patch string to `tool_input.command` and JSON-parsing `toolArgs` for `edit`/`create` (→ `tool_input.file_path`); emit the `additionalContext` payload. Degrades to a safe no-op on malformed input.
- **`skill/scripts/hook-admin.mjs` / `cli/bin/commands/skills.mjs`** — install + idempotent-repair `.github/hooks/impeccable.json` (bash-aware marker stripping).
- **`skill/reference/hooks.md`** — document GitHub Copilot as a supported harness.
- **Tests** — builder shape, routing, harness detection, event normalization (edit + create + **apply_patch** + malformed `toolArgs`), and end-to-end `runHook` for both the `-p` and interactive/cloud paths.
- **`.github/hooks/impeccable.json`** — generated root manifest (committed alongside its sibling root manifests).

## Verification

Captured the real hook I/O from a trusted interactive Copilot session:

```
Copilot → hook:  toolName:"apply_patch", toolArgs:"*** Begin Patch\n*** Add File: _hooktest.css\n+body { font-family: 'Inter'... }\n*** End Patch"
hook → Copilot:  {"additionalContext":"[impeccable@1] Design hook findings requiring review (3 issue(s)):
                  - L1 [overused-font] ... Inter ...
                  - L4 [bounce-easing] ... cubic-bezier ...
                  - L1 [design-system-font] ... "}
```

The hook fired on the edit, parsed the patch, ran the detector, and returned a correct `additionalContext` reminder. (`copilot -p` runs untrusted and skip repo hooks — firing requires a trusted session, which is the documented behavior.)

## Notes

- Source-first per repo policy: generated provider skill permutations (`.claude/skills/…`, `.gemini/…`, `plugin/…`) are **not** staged here — `sync-generated-output.yml` regenerates them post-merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Touches shared hook runtime and install merge logic across harnesses; Copilot’s distinct event schema and apply_patch handling add parsing edge cases, though coverage is broad in new tests.
> 
> **Overview**
> Adds **GitHub Copilot** as a fourth harness for the Impeccable design detector hook, matching the post-edit reminder flow already used by Claude Code and Codex (Cursor stays pre-edit).
> 
> Install and build now emit a committed repo manifest at **`.github/hooks/impeccable.json`** (`postToolUse`, flat `bash`/`timeoutSec` entries, matcher `edit|create|apply_patch`). The CLI **`skills install`** path treats `.github` like other providers but **does not rewrite** hook commands to machine-local paths, so Copilot CLI and cloud agents keep a portable `git rev-parse` script path.
> 
> **`hook-lib.mjs`** gains a `github` harness: auto-detect from Copilot’s camelCase `toolName`/`toolArgs`, normalize JSON-string args for `edit`/`create` vs raw patch strings for **`apply_patch`**, and return hook output as top-level **`additionalContext`** instead of `hookSpecificOutput`. Manifest install/repair and Impeccable hook stripping now recognize **`bash`/`powershell`** keys. Docs, changelog, and tests cover the new manifest builder, install path, and end-to-end `runHook` for both edit and apply_patch events.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 476586cf7ebc9182175547e73646ebcba9ff92e9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: skill/scripts/hook-lib.mjs [hunk 1]
```diff
@@ -959,13 +959,104 @@ export function resolveTargetFiles(event, projectCwd) {
...
+    const patch = applyPatchText(rawArgs);
+    if (patch) {
+      toolInput.command = patch;
+      normalizedToolName = 'apply_patch';
+    }
+  } else {
+    const args = parseGitHubToolArgs(rawArgs);
+    const filePath = args.path || args.file_path || args.filePath || args.target_file;
+    if (typeof filePath === 'string' && filePath) toolInput.file_path = filePath;
+  }
```
[the remark this answers is not in the document]
Comment (pbakaus) [reply skill/scripts/hook-lib.mjs:1055]: Fixed in `476586cf7ebc9182175547e73646ebcba9ff92e9`. `looksLikeApplyPatch` now only treats a `toolArgs` string as `apply_patch` when it does **not** parse as a JSON object. A real `apply_patch` payload is a raw patch string (won't parse as JSON); an `edit`/`create` payload is a JSON object whose edited *content* may merely contain patch markers, so it keeps the JSON path and extracts `path`. Added a regression test (`does not misroute an edit whose content contains apply_patch markers`).

File: cli/bin/commands/skills.mjs [hunk 2]
```diff
@@ -1160,7 +1166,10 @@ function valueHasImpeccableHookMarker(value) {
 
 function stripImpeccableHookEntry(entry) {
   if (!entry || typeof entry !== 'object') return entry;
-  if (valueHasImpeccableHookMarker(entry.command) || valueHasImpeccableHookMarker(entry.args)) {
```
[the remark this answers is not in the document]
Comment (pbakaus) [reply cli/bin/commands/skills.mjs:1163]: Intentional, now documented in `476586cf7ebc9182175547e73646ebcba9ff92e9`. `.github/hooks/impeccable.json` is a **committed, team-shared** manifest that the Copilot cloud agent and every teammate read, so the command must stay portable via `$(git rev-parse --show-toplevel)/.github/skills/...`. Rewriting it to a machine-local absolute `skillRoot` path (as a `.github` branch in `hookScriptPathForProvider` would do) would break the cloud agent and other machines. GitHub skills are project-scoped (not in the home-provider set), so the project-relative path always resolves. Added a comment explaining the deliberate exclusion.
